module/x/gravity/keeper/slashing.go

```diff
@@ -8,16 +8,59 @@ import (
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 )
 
+type ValidatorSlashingInfo struct {
+    Validator   stakingtypes.Validator
+    Exists bool
+    SigningInfo  slashingtypes.ValidatorSigningInfo
+    ConsAddress  sdk.ConsAddress
+}
+
+// GetUnbondingValidatorSlashingInfos returns the information needed for slashing for each unbonding validator 
+func (k Keeper) GetUnbondingValidatorSlashingInfos(ctx sdk.Context) ([]stakingtypes.Validator, []ValidatorSlashingInfo) {
+	blockTime := ctx.BlockTime().Add(k.StakingKeeper.GetParams(ctx).UnbondingTime)
+	blockHeight := ctx.BlockHeight()
+
+    var unbondingValInfos []ValidatorSlashingInfo	
+    var unbondingValidators []stakingtypes.Validator
+    unbondingValIterator := k.StakingKeeper.ValidatorQueueIterator(ctx, blockTime, blockHeight)
+	defer unbondingValIterator.Close()
+	for ; unbondingValIterator.Valid(); unbondingValIterator.Next() {
+		unbondingValidatorsAddr := k.GetUnbondingValidators(unbondingValIterator.Value()).Addresses
```

Comment:
I looked at `GetUnbondingValidators` and it doesn't seem to be doing what it's describing at all? Should we just delete that function from the keeper and move some of its bits in here if it isn't being used elsewhere?

Reply:
I could just rename it?

Reply:
If what it's doing is only necessary in ABCI it would be cleaner to just pull out the necessary lines and delete it.

Reply:
Let's also delete the function from the keeper since this would have been the only use.